tests/unit/api/multiObjectDelete.js

```diff
@@ -134,54 +155,151 @@ describe('getObjMetadataAndDelete function for multiObjectDelete', () => {
+                              '</Grantee>' +
+                              '<Permission>WRITE</Permission>' +
+                            '</Grant>' +
+                          '</AccessControlList>' +
+                        '</AccessControlPolicy>',
+                    url: '/?acl',
+                    query: { acl: '' },
+                };
+                bucketPutACL(authInfo, testACLRequest, log, () => {
+                    objectPut(authInfo, testPutObjectRequest1, undefined,
+                    log, () => {
+                        objectPut(altAuthInfo, testPutObjectRequest2, undefined,
+                        log, () => {
+                            objectPut(altAuthInfo, testPutObjectRequest3,
+                            undefined, log, () => {
+                                assert.strictEqual(metadata.keyMaps
+                                    .get(bucketName)
+                                    .has(objectKey1), true);
+                                assert.strictEqual(metadata.keyMaps
+                                    .get(bucketName)
+                                    .has(objectKey2), true);
+                                assert.strictEqual(metadata.keyMaps
+                                    .get(bucketName)
+                                    .has(objectKey3), true);
+                                done();
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        it('should successfully get object metadata and then ' +
+        'delete metadata and data', done => {
+            getObjMetadataAndDelete(authInfo, authInfo.getCanonicalID(),
+            request, bucketName, bucket, true, [],
+            [{ key: objectKey1 }, { key: objectKey2 }, { key: objectKey3 }],
+            log, (err, quietSetting, errorResults, deletedObjStats) => {
+                assert.ifError(err);
+                assert.strictEqual(quietSetting, true);
+                assert.deepStrictEqual(errorResults, []);
+                assert.strictEqual(deletedObjStats.requesterIsObjOwner.
+                    numOfObjectsRemoved, 1);
+                assert.strictEqual(deletedObjStats.requesterNotObjOwner.
+                    numOfObjectsRemoved, 2);
+                assert.strictEqual(deletedObjStats.requesterIsObjOwner.
+                    totalContentLengthDeleted, postBody.length);
+                assert.strictEqual(deletedObjStats.requesterNotObjOwner.
+                    totalContentLengthDeleted, contentLength);
+                assert.strictEqual(metadata.keyMaps.get(bucketName)
+                    .has(objectKey1), false);
+                assert.strictEqual(metadata.keyMaps.get(bucketName)
+                    .has(objectKey2), false);
+                assert.strictEqual(metadata.keyMaps.get(bucketName)
+                    .has(objectKey3), false);
+                // call to delete data is async so wait 20 ms to check
```

Comment:
20ms sounds rather short to me even if it's just to wait some node.js scheduling, maybe 200ms would reduce the chances of being flaky? Not sure just an impression since the tests will run on potentially heavily loaded environments where slowdowns are common.

Reply:
It does seem short lol, but I borrowed that value from the existing test for deletions, and I don't think there have been any issues with flakiness, so I think it should be ok